test/utils/block_reward_functions.js

```diff
@@ -0,0 +1,93 @@
+    3: '1719356152662202255',
+    5: '2865593587770337092',
+    100: '57311871755406741844'
+  };
+
+  it('Get correct reward value for the entered rate', async () => {
+    expect(convertRateToBaseReward(0.01)).to.equal(resultsToInputs[0.01]);
+    expect(convertRateToBaseReward(0.03)).to.equal(resultsToInputs[0.03]);
+    expect(convertRateToBaseReward(0.05)).to.equal(resultsToInputs[0.05]);
+    expect(convertRateToBaseReward(0.1)).to.equal(resultsToInputs[0.1]);
+    expect(convertRateToBaseReward(0.3)).to.equal(resultsToInputs[0.3]);
+    expect(convertRateToBaseReward(0.5)).to.equal(resultsToInputs[0.5]);
+    expect(convertRateToBaseReward(1)).to.equal(resultsToInputs[1]);
+    expect(convertRateToBaseReward(1.01)).to.equal(resultsToInputs[1.01]);
+    expect(convertRateToBaseReward(1.3)).to.equal(resultsToInputs[1.3]);
+    expect(convertRateToBaseReward(2)).to.equal(resultsToInputs[2]);
+    expect(convertRateToBaseReward(2.5)).to.equal(resultsToInputs[2.5]);
+    expect(convertRateToBaseReward(3)).to.equal(resultsToInputs[3]);
+    expect(convertRateToBaseReward(5)).to.equal(resultsToInputs[5]);
+  });
+
+  it ('Get correct reward value for extreme rate values', async () => {
+    expect(convertRateToBaseReward(0)).to.equal(resultsToInputs[0]);
+    expect(convertRateToBaseReward(100)).to.equal(resultsToInputs[100]);
+  });
+
+  it('Get exception for the rate more than 100%', async () => {
+    expect(() => {
+      convertRateToBaseReward(101);
+    }).to.throw('Invalid rate');
+  });
+
+  it('Get exception for the negative rate', async () => {
+    expect(() => {
+      convertRateToBaseReward(-2);
+    }).to.throw('Invalid rate');
+  });
+
+  it('Get correct rate calculated for reward value', async () => {
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[0.01]))).to.equal(0.01);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[0.03]))).to.equal(0.03);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[0.05]))).to.equal(0.05);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[0.1]))).to.equal(0.1);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[0.3]))).to.equal(0.3);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[0.5]))).to.equal(0.5);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[1]))).to.equal(1);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[1.01]))).to.equal(1.01);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[1.3]))).to.equal(1.3);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[2]))).to.equal(2);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[2.5]))).to.equal(2.5);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[3]))).to.equal(3);
+    expect(parseFloat(convertBaseRewardToRate(resultsToInputs[5]))).to.equal(5);
+  });
+
+  it ('Roundtrip rate test', async () => {
+    const defaultRate = (Math.random() * 100).toFixed(2);
```

Comment:
Select a representative test case(s). Or run for the `resultsToInputs`. Or use property-based testing. But remove random. Consider that if the test failed the best we can do is take the value for which it failed, add a new test case and correct the bug.

Reply:
Fixed